### PR TITLE
Add Uptime Command.

### PR DESCRIPTION
**Pull request by RealYusufIsmail**
This will give you the ability to see how long the bot has been up for.

### CODE REVIEW

**Comment by Tais993** [comment]
Do you have pictures on how it looks for the user?

**Comment by Zabuzard** [comment]
Whats the purpose of this command? Imo this just clutters the list of available commands with a command that is pretty much useless.

Voting to close.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.internal.utils.Checks;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+
+import java.lang.management.ManagementFactory;
+
+/**
+ * Informs the use how long the bot has been running for.
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:13]: wrong english. also, elaborate.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ */
+public class UpTimeCommand extends SlashCommandAdapter {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:15]: final

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    /**
+     * Constructs an instance.
+     */
+    public UpTimeCommand() {
+        super("uptime", "Tells the user how long the bot has been up for",
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:20]: the description is bad UX, this is talking to the user and not to the programmer. rewrite

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+
+        String uptime = (years == 0 ? "" : "**" + years + "** Years, ")
+                + (months == 0 ? "" : "**" + months + "** Months, ")
+                + (days == 0 ? "" : "**" + days + "** Days, ")
+                + (hours == 0 ? "" : "**" + hours + "** Hours, ")
+                + (minutes == 0 ? "" : "**" + minutes + "** Minutes, ")
+                + (seconds == 0 ? "" : "**" + seconds + "** Seconds, ");
+        uptime = replaceLast(uptime, ", ", "");
+        uptime = replaceLast(uptime, ",", " and");
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:43]: Use modern java please. `java.time` API got you covered.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+     * Constructs an instance.
+     */
+    public UpTimeCommand() {
+        super("uptime", "Tells the user how long the bot has been up for",
+                SlashCommandVisibility.GUILD);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        final long duration = ManagementFactory.getRuntimeMXBean().getUptime();
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:26]: The frick is this. Dont. instead, remember the time during startup and compute the difference to that.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        uptime = replaceLast(uptime, ",", " and");
+
+        event
+            .replyEmbeds(new EmbedBuilder().setTitle("Uptime")
+                .setDescription("I have been up for " + uptime + ".")
+                .build())
+            .queue();
+    }
+
+    public static @NotNull String replaceLast(String text, String search, String replacement) {
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:52]: NotNull on all params

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        // Search not found
+        if (index == -1) {
+            return text;
+        }
+
+        final String firstPart = text.substring(0, index);
+        final String lastPart = text.substring(index + search.length());
+
+        return firstPart + replacement + lastPart;
+    }
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:68]: whats the purpose of this method? I dont think its needed at all, at least if u use proper `java.time`.

File: application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java (hunk 6 above)
Comment (Zabuzard) [reply application/src/main/java/org/togetherjava/tjbot/commands/basic/UpTimeCommand.java:52]: public methods need full javadoc. also, a method like this should be covered by a full unit-test

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Pull request closed without merging**